lib/string.js

```diff
@@ -292,18 +292,56 @@ internals.String = class extends Any {
         });
     }
 
-    guid() {
+    guid(guidOptions) {
 
-        const regex = /^[A-F0-9]{8}(?:-?[A-F0-9]{4}){3}-?[A-F0-9]{12}$/i;
-        const regex2 = /^\{[A-F0-9]{8}(?:-?[A-F0-9]{4}){3}-?[A-F0-9]{12}\}$/i;
+        // 89AB on last group of 4
+        // Version on second group of 4
+
+        const brackets = {
+            '{': '}', '[': ']', '(': ')', '': ''
+        };
+
+        const uuids = {
+            'uuidv1': '1',
+            'uuidv2': '2',
+            'uuidv3': '3',
+            'uuidv4': '4',
+            'uuidv5': '5'
+        };
+
+        const regex = /^([\[{\(]?)([0-9A-F]{8})([:-]?)([0-9A-F]{4})([:-]?)([0-9A-F]{4})([:-]?)([0-9A-F]{4})([:-]?)([0-9A-F]{12})([\]}\)]?)$/i;
```

Comment:
We could gain some additional performance on this Regex by switching groups that we don't need the result from to non-capture groups. For example, I don't see us using the separator group `([:-]?)` so switching that to a non-capture group `(?:[:-]?)` usually gains a few hundred thousand operations/second in my tests.